### PR TITLE
PIQA Dataset - Two subtasks

### PR DESCRIPTION
**Pull request by kuntalkumarpal**

### CODE REVIEW

**Review by danyaljj — CHANGES_REQUESTED** [review]
Thanks! 

- Please update the readme that contains the task names and descriptions. 
- Please rephrase the content of the "definition" field such that it contains only the necessary details.

**Comment by kuntalkumarpal** [comment]
Changes made : 
1. Rephrased the definition of the tasks removing unnecessary comments.
2. Updated the task readme file

**Comment by danyaljj** [comment]
Thanks! 
 - I still think that the content could be more direct: 
  * `These are easy for humans but hard for the machines.`: this is an underdefined statement. I would drop it. 
  * `Here you are given a goal which describes what do you want to achieve.`: not sure what this sentence is saying. A "goal", by definition, is something that we "want to achieve". I think what you want to say is, something along the line of: "Given the provided goal task in the input, describe a process that would lead to the asked outcome. This process often involves physical motions with objects, such as moving  them, arranging them in a certain way, mixing them, shaking them,  etc."  

 - Please update the branch with master.

**Comment by danyaljj** [comment]
One more comment: make sure that your email [is set in your git environment and it is also mentioned in your github profile](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address).

**Comment by kuntalkumarpal** [comment]
Thank you for your detailed comments.

- Updated the task definitions. Please let me know if there are any issues
- Resolved the merge conflict in the tasks/README.md
- Git commits are now connected with the correct email address

**Comment by danyaljj** [comment]
Thanks! 

A couple more comments: 

- Not sure what the following means: `It is important to note that you should not use terms or references that require background knowledge.` How can we make it more clear? 

I am not happy with the explanations of the examples. Here are several examples why: 
- in multiple places: `but fool the machines` --> irrelevant?  
- `Irrelevant Trick: The trick is too easy` -> Unclear. What does `trick` mean here? (a solution? a response?) 
- `The question is stand-alone` --> redundant?

**Review by swarooprm — COMMENTED** [review]
Please pull and merge with the main branch. That should trigger the new automatic tests.

**Comment by kuntalkumarpal** [comment]
- Updated the explanations of the examples
- Removed conflicts of readme
- Fixed repeated input string issue

**Review by swarooprm — COMMENTED** [review]
Looks good to me! Approve from my side.

File: tasks/README.md [hunk 1]
```diff
@@ -67,13 +67,16 @@ Name | Summary | Category
 `task060_ropes_question_generation`	| Constructing questions regarding relations in the given paragraph.	| Question Generation
 `task061_ropes_answer_generation`	| Answering questions regarding relations in the given paragraph.	| Answer Generation
 `task062_bigbench_repeat_copy_logic`	|  Generating text that follows simple logical operations such as "repeat", "before", "after" etc.	| Logic
+`task073_CommonsenseQA_answer_generation`	|  Answering questions based on commonsense knowledge.	| Answer Generation
```
Comment (danyaljj) [inline tasks/README.md]: is this redundant?

File: tasks/README.md [hunk 2]
```diff
@@ -67,13 +67,16 @@ Name | Summary | Category
...
 `task073_CommonsenseQA_answer_generation` | Answering questions based on commonsense knowledge | Answer Generation
 `task074_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
 `task075_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation  
 `task076_splash_correcting_sql_mistakes` | Based on feedback correct the mistake in a given SQL statement. | Structured Query Generation, Text Modification
 `task077_splash_explanation_to_sql` | Generate an SQL statement based on a description of what the SQL statement does. | Structured Query Generation
 `task078_splash_sql_to_explanation` | Give a natural language description of what a given SQL statement is doing. | Structured Query Classification
 `task079_conala_concat_strings` | Given a list of strings concatenate them to form one string | Answer Generation
+`task080_piqa_answer_generation`	|  Generating solution to a goal regarding physical knowledge about the world	| Answer Generation
+`task081_piqa_wrong_answer_generation`	|  Generating incorrect solution to a goal regarding physical knowledge about the world	| Incorrect Answer Generation
```
Comment (danyaljj) [inline tasks/README.md:86]: wrong -> incorrect

**Review by danyaljj — CHANGES_REQUESTED** [review]
I am not happy with the instructions for the incorrect answer generation; the description does not seem concrete. For example, what does `obvious incorrect answers` mean? (in `Also avoid obvious incorrect answers.`) We should find a better way to phrase these. Is it how the task was described in the original template?

File: tasks/README.md (hunk 2 above)
Comment (kuntalkumarpal) [reply tasks/README.md:86]: Should I use wrong or incorrect ?  In MCTACO Tasks in the Name it is 'wrong' and in the category it is 'incorrect'.

File: tasks/README.md (hunk 1 above)
Comment (kuntalkumarpal) [reply tasks/README.md]: For the statement _"obvious incorrect answer"_, would adding examples make it clear like "Avoid obvious incorrect answers like cooking with motor oil or drying keyboard keys with wet cloth" ?

File: tasks/README.md (hunk 2 above)
Comment (danyaljj) [reply tasks/README.md:86]: Alright, let's ignore my comment then.

**Comment by danyaljj** [comment]
> For the statement "obvious incorrect answer", would adding examples make it clear like "Avoid obvious incorrect answers like cooking with motor oil or drying keyboard keys with wet cloth" ? 


@swarooprm thoughts on how to make the incorrect answer generation a bit more concrete?

**Comment by danyaljj** [comment]
@kuntalkumarpal are you using the task templates? (in which case, could you share a screenshot of their task for 'incorrect answer generation'?)

**Comment by kuntalkumarpal** [comment]
@danyaljj I used the trick statement of the "good" examples (image2) for incorrect answer generation examples. I have manually written the explanations for them based on the red underlined keywords since there are no explanations present.

**Comment by swarooprm** [comment]
> @swarooprm thoughts on how to make the incorrect answer generation a bit more concrete?

I would extract these lines from the template and add something like this in the existing definition you have: "Write a logical but wrong solution to the given question. Make sure that your incorrect answer is not obvious to eliminate as a contender for the correct answer to the given question. An incorrect answer is obvious to eliminate if it introduces objects or events unrelated to the context of the question" e.g. "cleaning keyboard by making an elephant sit on them" or if it involves contradictory statements e.g. "drying wet keyboards with running water"."

**Comment by kuntalkumarpal** [comment]
Updated Incorrect answer generation description as suggested by @swarooprm

**Comment by danyaljj** [comment]
- "would not lead to the asked outcome.": need to be expanded: `would not lead to the asked outcome, yet ....` 
- "incorrect answer is not obvious to eliminate as a contender for the correct answer" --> there is no notion of the multiple-choice answer at this level. What does "eliminate" even mean in this context?
 - "logical but incorrect": unclear phrasing 



Here is a rephrasing of the task definition: 

> Given the provided goal task in the input, describe a physical process (such as moving them, arranging them in a certain way, mixing them, shaking them, etc) that does not lead to the asked outcome, yet it is closely related to it (i.e., it should use the words that are provided in the input). To make sure that the generated process does not lead to the asked outcome. you can introduce objects or events unrelated to the context of the question e.g. "cleaning the keyboard by making an elephant sit on them"; or you can create contradictory statements e.g. "drying wet keyboards with running water".  In your responses, avoid typos and misspellings.

**Comment by danyaljj** [comment]
@kuntalkumarpal There is some feedback from crowd workers regarding your task(s), summarized here: 
https://github.com/allenai/natural-instructions-expansion/pull/276#issuecomment-942714679
Wondering if you can address the feedback?

**Comment by kuntalkumarpal** [comment]
> @kuntalkumarpal There is some feedback from crowd workers regarding your task(s), summarized here: [#276 (comment)](https://github.com/allenai/natural-instructions-expansion/pull/276#issuecomment-942714679) Wondering if you can address the feedback?

@danyaljj I have addressed this in  PR #476 by adding three more examples each. Please have a look and let me know if it looks ok

**Comment by danyaljj** [comment]
Thanks!

**Pull request merged**